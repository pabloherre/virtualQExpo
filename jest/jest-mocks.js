import mockAsyncStorage from '@react-native-community/async-storage/jest/async-storage-mock';

const mockfeathersMethods = {
  create: jest.fn(),
  update: jest.fn(),
  find: jest.fn(),
  get: jest.fn(),
  patch: jest.fn(),
  remove: jest.fn()
};
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.requireActual('./mocks');
jest.mock('@react-native-community/async-storage', () => mockAsyncStorage);
jest.mock('expo-barcode-scanner');
jest.mock('expo-location');
jest.mock('react-native-maps');
jest.mock('../src/setup/feathersClient', () => {
  return {
    __esModule: true,
    default: jest.fn(),
    userApi: mockfeathersMethods,
    appointmentApi: mockfeathersMethods,
    businessApi: mockfeathersMethods,
    categoryApi: mockfeathersMethods,
    turnApi: mockfeathersMethods,
    authenticate: jest.fn(),
    reAuthenticate: jest.fn()
  };
});
jest.mock('../src/services/user/User.service');
jest.mock('../src/modules/auth/Auth.service');
jest.mock('../src/common/sideMenu/SideMenu.service');
jest.genMockFromModule('../src/setup/store');
