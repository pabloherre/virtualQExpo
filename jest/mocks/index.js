export * from './appointment.mock';
export * from './appointments.mock';

