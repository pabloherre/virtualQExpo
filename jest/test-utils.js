import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

const mockStore = configureStore([]);
function connectedRender(ui, initialState) {
  const store = mockStore(initialState);
  component = renderer.create(<Provider store={store}>{ui}</Provider>);
  return component;
}

// override connectedRender method
export { connectedRender };
