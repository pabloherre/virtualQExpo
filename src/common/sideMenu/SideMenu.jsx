import React from 'react';
import { connect } from 'react-redux';
import RoundedButton from '../buttons/RoundedButton/RoundedButton';
import { View } from 'react-native';
import AuthService from '../../modules/auth/Auth.service';
import UserService from '../../services/user/User.service';
import { colors } from '../../../theme';
import { FontAwesome, SimpleLineIcons } from '@expo/vector-icons';
import { SideMenuService } from './SideMenu.service';
import Typography from '../typography/Typography';
import { setTestInfo } from '../../utils/test.utils';

export class Menu extends React.Component {
  onPress = async () => {
    SideMenuService.closeMenu();
    await AuthService.logout();
    await UserService.setUser(null);
    this.props.navigation.navigate('Login');
  };

  render() {
    return (
      <View
        style={{
          flex: 1,
          alignItems: 'stretch',
          backgroundColor: colors.card
        }}
      >
        <View
          style={{
            alignItems: 'center',
            borderBottomWidth: 1,
            paddingVertical: 20,
            marginHorizontal: 20,
            borderBottomColor: colors.border
          }}
        >
          <FontAwesome name="user-circle" size={120} color={colors.secondary} style={{ paddingBottom: 20 }} />
          <Typography color="secondary" size="small">
            {this.props.user && this.props.user.email}
          </Typography>
        </View>
        <View style={{ flex: 1, marginVertical: 10, width: 200, alignSelf: 'center', justifyContent: 'flex-end' }}>
          <RoundedButton
            startIcon={<SimpleLineIcons name="logout" size={24} color="white" />}
            label="Logout"
            onPress={this.onPress}
            {...setTestInfo('testTouchableLogout')}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = state => {
  return {
    user: state.user.data
  };
};

export default connect(mapStateToProps, null)(Menu);
