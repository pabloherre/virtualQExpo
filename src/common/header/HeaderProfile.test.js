import React from 'react';
import ConnectedHeaderProfile, { HeaderProfile } from './HeaderProfile';
import { shallow } from 'enzyme';
import AuthService from '../../modules/auth/Auth.service';
import { SideMenuService } from '../sideMenu/SideMenu.service';
import { connectedMount } from '../../../jest/test-utils';

let component;

const mockProps = {
  navigation: {
    push: jest.fn()
  }
};

describe('<HeaderProfile />', () => {
  beforeEach(() => {
    component = shallow(<HeaderProfile {...mockProps} />);
  });
  afterEach(() => {
    mockProps.navigation.push.mockClear();
    AuthService.logout.mockClear();
    SideMenuService.openMenu.mockClear();
    SideMenuService.closeMenu.mockClear();
  });
  // it('renders correctly', async () => {
  //   expect(component).toMatchSnapshot();
  // });

  it('should have an icon', async () => {
    expect(component.find('Icon')).toExist();
  });

  it('can receive a color for the icon and should render it', async () => {
    component.setProps({ colors: { text: 'red' } });

    expect(component.instance().props.colors.text).toBe('red');
    expect(component).toMatchSnapshot();
  });

  it('should render a default color if none passed', async () => {
    expect(component.instance().props.colors.text).toBe('#a1a2a4');
    expect(component).toMatchSnapshot();
  });

  it('it calls to open side menu when press button', async () => {
    const component = connectedMount(<ConnectedHeaderProfile />, { sideMenu: { open: false } });
    await component.childAt(0).childAt(0).childAt(0).childAt(0).props().onPress();
    expect(SideMenuService.openMenu).toHaveBeenCalled();
  });

  it('it calls to close side menu when press button', async () => {
    const component = connectedMount(<ConnectedHeaderProfile />, { sideMenu: { open: true } });
    await component.childAt(0).childAt(0).childAt(0).childAt(0).props().onPress();
    expect(SideMenuService.closeMenu).toHaveBeenCalled();
  });
});
