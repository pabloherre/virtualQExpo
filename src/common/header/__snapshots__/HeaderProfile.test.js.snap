// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderProfile /> can receive a color for the icon and should render it 1`] = `
<Icon
  accessibilityLabel="testTouchableIconSideMenuToggle"
  allowFontScaling={false}
  color="red"
  name="user"
  onPress={[Function]}
  size={36}
  testID="testTouchableIconSideMenuToggle"
/>
`;

exports[`<HeaderProfile /> should render a default color if none passed 1`] = `
<Icon
  accessibilityLabel="testTouchableIconSideMenuToggle"
  allowFontScaling={false}
  color="#a1a2a4"
  name="user"
  onPress={[Function]}
  size={36}
  testID="testTouchableIconSideMenuToggle"
/>
`;
